src/lib/dnssd/IncrementalResolve.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <lib/dnssd/IncrementalResolve.h>
+
+#include <lib/dnssd/ServiceNaming.h>
+#include <lib/dnssd/TxtFields.h>
+#include <lib/dnssd/minimal_mdns/core/RecordWriter.h>
+#include <lib/support/CHIPMemString.h>
+#include <trace/trace.h>
+
+namespace chip {
+namespace Dnssd {
+
+using namespace mdns::Minimal;
```

Comment:
This file looks pretty specific to minimal mdns, but that's very non-obvious from the filename, unlike most of the other backend-specific files in this directory.